src/sonic-yang-models/yang-models/sonic-nat.yang

```diff
@@ -0,0 +1,316 @@
+                            error-app-tag nat-tcp-timeout-invalid;
+                        }
+                    }
+                }
+                leaf nat_udp_timeout {
+                    description "Timeout for the nat udp entries within the range of 120 sec to 600 secs.";
+
+                    type uint16 {
+                        range "120..600" {
+                            error-message "Invalid NAT UDP Timeout, valid range 120 sec to 600 sec.";
+                            error-app-tag nat-udp-timeout-invalid;
+                        }
+                    }
+                }
+            }
+        }
+
+        container NAT_POOL {
+
+            description "NAT_POOL table part of config_db.json";
+
+            list NAT_POOL_LIST {
+                key "name";
+                max-elements 16;
+
+                leaf name {
+                    description "Key - Name of the NAT Pool";
+
+                    type string {
+                        pattern '[a-zA-Z0-9]{1}([-a-zA-Z0-9_]{0,31})';
+                        length 1..32 {
+                            error-message "Invalid length for the pool name.";
+                            error-app-tag pool-name-invalid-length;
+                        }
+                    }
+                }
+
+                leaf nat_ip {
+                    description "Single IP address or a range of addresses for a NAT pool.";
+
+                    mandatory true;
+
+                    type ip-address-range;
+                }
+
+                leaf nat_port {
+                    description "Range of port values for a NAT pool.";
+
+                    type string {
+                        pattern "(([0-9]{1,4}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-4])(-)([0-9]{1,4}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5]))";
+                    }
+                }
+            }
+        }
+
+        container NAT_BINDINGS {
+
+            description "NAT_BINDINGS table part of config_db.json";
+
+            list NAT_BINDINGS_LIST {
+                key "name";
+                max-elements 16;
+
+                leaf name {
+                    description "Key - Name of the NAT Binding";
+
+                    type string {
+
+                        pattern '[a-zA-Z0-9]{1}([-a-zA-Z0-9_]{0,31})';
+                        length 1..32 {
+                            error-message "Invalid length for the binding name.";
+                            error-app-tag binding-name-invalid-length;
+                        }
+                    }
+                }
+
+                leaf nat_pool {
+                    description "NAT Pool name mapping for the binding";
+
+                    mandatory true;
+
+                    type leafref {
+                        path "../../../NAT_POOL/NAT_POOL_LIST/name";
+                    }
+                }
```

Comment:
We need to update DO_NOT_NAT_ACTION in ACL Yang model?

Reply:
Done, added DO_NOT_NAT as action.